src/address.c

```diff
@@ -0,0 +1,73 @@
+#include "internal.h"
+#include "base58.h"
+#include "ccan/ccan/build_assert/build_assert.h"
+#include <include/wally_address.h>
+#include <include/wally_bip32.h>
+#include <include/wally_crypto.h>
+#include <include/wally_script.h>
+
+int wally_bip32_key_to_address(const struct ext_key *hdkey, uint32_t flags,
+                               uint32_t version, char **output)
+{
+    unsigned char address[HASH160_LEN + 1];
+    int ret;
+
+    if (output)
+        *output = NULL;
+
+    if (!hdkey || (version & ~0xff) || (flags & ~0xff) || !output)
+        return WALLY_EINVAL;
+
+    if (!(flags & (WALLY_ADDRESS_TYPE_P2PKH | WALLY_ADDRESS_TYPE_P2SH_P2WPKH)))
+        return WALLY_EINVAL;
+
+    // Catch known incorrect combinations of address type and version:
+    if ((flags & WALLY_ADDRESS_TYPE_P2PKH && version == 0x05) ||
+        (flags & WALLY_ADDRESS_TYPE_P2SH_P2WPKH && version == 0x00) ||
+        (flags & WALLY_ADDRESS_TYPE_P2PKH && version == 0xC4) ||
+        (flags & WALLY_ADDRESS_TYPE_P2SH_P2WPKH && version == 0x6F))
```

Comment:
Maybe it's worth defining these constants (0, 5, 111, 196) especially as they are being used in the related PR: https://github.com/ElementsProject/libwally-core/pull/112

Reply:
Done